core/server/common/src/main/java/alluxio/master/MasterProcess.java

```diff
@@ -173,22 +181,19 @@ void registerServices(GrpcServerBuilder serverBuilder,
     }
   }
 
-  /**
-   * Waits until the grpc server is ready to serve requests.
-   *
-   * @param timeoutMs how long to wait in milliseconds
-   * @return whether the grpc server became ready before the specified timeout
-   */
-  public boolean waitForGrpcServerReady(int timeoutMs) {
+  @Override
+  public boolean waitForReady(int timeoutMs) {
+    if (mLeaderSelector instanceof UFSJournalSingleMasterPrimarySelector
```

Comment:
Only after the `UFSJournalSingleMasterPrimarySelector#start` method is called, which is why the distinction is important. There is a brief window between the start of `AlluxioMasterProcess#start` and when `mLeaderSelector.start` is called where it will return `STANDBY`. 
This conforms to the design that we've chosen where masters are first started in `STANDBY` mode, and then gain primacy later. However, we must account for the fact that a single UFS master is not ready during this phase: a single UFS master is ready once its GRPC server is up.

Reply:
okay makes sense can you add a comment for this? Thanks!

Reply:
I will wait for other reviewers to add their comments and address them all at once.

Reply:
I think this creates a data race on mServingThread.

Also it looks like this is correct to from within the master process start() thread, but will be unsafe in other places.

I think the interface it is overriding is meant to be called from anywhere, and seems to be used in some tests, so I think we have to assume this and make it safe to be run from any state.

Reply:
+1 on double checking the race